### PR TITLE
Update the install-fabric script & docs

### PR DESCRIPTION
**Pull request by mbwhite**
- Add podman to install-fabric script
- Add documentation on this script

Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

### CODE REVIEW

File: docs/source/install.md [hunk 1]
```diff
@@ -74,6 +74,58 @@ curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.4.2 1.5.2
...
+
+```bash
+curl -sSLO https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/install-fabric.sh && chmod +x install-fabric.sh
+```
+
+Run the script with the `-h` option to see the options
+
+```bash
+./install-fabric.sh -h
+Usage: ./install-fabric.sh [-f|--fabric-version <arg>] [-c|--ca-version <arg>] <comp-1> [<comp-2>] ... [<comp-n>] ...
```
Comment (nikhil550) [inline docs/source/install.md:44]: The only nit I would have is that I found this help text confusing. For example, this line :`d[ocker]|b[inary]|s[amples]` is very ugly and hard to parse.

**Review by nikhil550 — DISMISSED** [review]
I ran the commands on my Intel Mac and everything ran fine.

File: docs/source/install.md (hunk 1 above)
Comment (mbwhite) [reply docs/source/install.md:44]: How about

```
<comp>: Component to install one or more of  docker | binary | samples | podman. (first letter of component also accepted)
```

Comment (nikhil550) [reply docs/source/install.md:44]: Much cleaner

Comment (mbwhite) [reply docs/source/install.md:44]: @nikhil550  thanks - I've updated the script accordingly

**Comment by mbwhite** [comment]
@hyperledger/fabric-core-maintainers could you review this please?  Thankyou

File: docs/source/install.md [hunk 2]
```diff
@@ -74,6 +74,58 @@ curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.4.2 1.5.2
   curl -sSL https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/bootstrap.sh| bash -s 
   ```
 
+
+
+# New Fabric Install Script
```
Comment (denyeart) [inline docs/source/install.md]: Seems redundant to document two different download/install scripts. Could we  simply replace the old instructions now and nudge people to use the new script only? The old instructions will remain in older release branches...

File: docs/source/install.md [hunk 3]
```diff
@@ -3,78 +3,85 @@
...
+To get the install script:
 
-```shell
-$ curl -sSL https://bit.ly/2ysbOFE | bash -s
+```bash
+curl -sSLO https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/install-fabric.sh && chmod +x install-fabric.sh
 ```
 
-You have completed installing Fabric samples, docker images, and binaries to your system.
+Run the script with the `-h` option to see the options
```
Comment (joshhus) [inline docs/source/install.md]: options:

File: docs/source/install.md [hunk 4]
```diff
@@ -3,78 +3,85 @@
...
+  * `peer`,
+  * `fabric-ca-client`,
+  * `fabric-ca-server`
 
+## Download Fabric samples, docker images, and binaries
 
-## Download Fabric samples, docker images, and binaries.
-
-Download `fabric-samples` to the `$HOME/go/src/github.com/<your_github_userid>` directory.  This is a Golang Community recommendation for Go projects. If you are using a different directory or Windows, see the [Notes](https://hyperledger-fabric.readthedocs.io/en/latest/install.html#notes) below.
+A working directory will be needed, As an example, Go Developers use the `$HOME/go/src/github.com/<your_github_userid>` directory.  This is a Golang Community recommendation for Go projects. If you are using a different directory or Windows, see the [Notes](https://hyperledger-fabric.readthedocs.io/en/latest/install.html#notes) below.
```
Comment (joshhus) [inline docs/source/install.md]: "A working directory is required&mdash;for example, Go developers use the ... "

File: docs/source/install.md [hunk 5]
```diff
@@ -3,78 +3,85 @@
...
+  * `cryptogen`,
+  * `discover`,
+  * `idemixgen`,
+  * `orderer`,
+  * `osnadmin`,
+  * `peer`,
+  * `fabric-ca-client`,
+  * `fabric-ca-server`
 
+## Download Fabric samples, docker images, and binaries
```
Comment (joshhus) [inline docs/source/install.md]: Docker is always capital Docker

**Comment by mbwhite** [comment]
@hyperledger/fabric-core-maintainers could you take a look please...

File: docs/source/install.md [hunk 6]
```diff
@@ -2,79 +2,86 @@
...
+```bash
+./install-fabric.sh docker samples
+./install-fabric.sh d s 
 ```
 
-## Notes
+If no arguments are supplied, then the arguments `docker binary samples` are assumed.
 
-### Other considerations
+## Chosing which version
```
Comment (andrew-coleman) [inline docs/source/install.md]:
```suggestion
## Choosing which version
```

**Review by andrew-coleman — APPROVED** [review]

File: scripts/install-fabric.sh [hunk 7]
```diff
@@ -50,9 +48,9 @@ begins_with_short_option()
 print_help()
 {
 	printf 'Usage: %s [-f|--fabric-version <arg>] [-c|--ca-version <arg>] <comp-1> [<comp-2>] ... [<comp-n>] ...\n' "$0"
-	printf '\t%s\n' "<comp>: Component to install one or more of  d[ocker]|b[inary]|s[amples]. If none specified, all will be installed"
-	printf '\t%s\n' "-f, --fabric-version: FabricVersion (default: '2.4.3')"
-	printf '\t%s\n' "-c, --ca-version: Fabric CA Version (default: '1.5.3')"
+	printf '\t%s\n' "<comp> Component to install, one or more of  docker | binary | samples | podman  First letter of component also accepted; If none specified docker | binary | samples is assumed"
```
Comment (denyeart) [inline scripts/install-fabric.sh:51]: Seeing this too late... but "podman" isn't a component to install, it is HOW to pull the images from docker hub, right?

File: scripts/install-fabric.sh [hunk 8]
```diff
@@ -268,29 +267,41 @@ CA_BINARY_FILE=hyperledger-fabric-ca-${ARCH}-${CA_VERSION}.tar.gz
...
         echo
         echo "Pull Hyperledger Fabric binaries"
         echo
         pullBinaries
 fi
 
-if [[ ${_arg_comp[@]} =~ d(ocker)? ]]; then
+if [[ "${_arg_comp[@]}" =~ ^p(odman)? ]]; then
+        echo
+        echo "Pull Hyperledger Fabric podman images"
```
Comment (denyeart) [inline scripts/install-fabric.sh:294]: They are not "podman images" or "docker images". They are images pulled using either docker command or podman command, right?

**Review by denyeart — COMMENTED** [review]
@mbwhite Sorry I didn't see this when it was opened/merged, but I'm not able to make sense of this as-is.

File: scripts/install-fabric.sh (hunk 8 above)
Comment (lehors) [reply scripts/install-fabric.sh:294]: that's correct

Comment (mbwhite) [reply scripts/install-fabric.sh:294]: Yes that's correct.... so it would be more accurate to say download the container images or not.. and what tool to choose.

Docker has  like 'hover' become a generic term for container/image... but is wrong...

File: scripts/install-fabric.sh (hunk 7 above)
Comment (mbwhite) [reply scripts/install-fabric.sh:51]: yes you are correct... see cmt below...
